pkg/main.go

```diff
@@ -19,25 +20,38 @@ func main() {
 	msgCh2 := make(chan messages.Message, bufLen)
 
 	bufLen = GetBufferLength("ACK_BUFFER_LENGTH")
-	ackCh := make(chan messages.Ack, bufLen)
+	ackCh1 := make(chan messages.Message, bufLen)
+	ackCh2 := make(chan messages.Message, bufLen)
 
-	reader, err := readers.NewReader(msgCh1, ackCh)
+	reader, err := readers.NewReader(msgCh1, ackCh1)
 	if err != nil {
 		panic(err)
 	}
 
-	writer, err := writers.NewWriter(msgCh2, ackCh)
+	writer, err := writers.NewWriter(msgCh2, ackCh2)
 	if err != nil {
 		panic(err)
 	}
 
-	middleware := middleware.NewMiddlewareManager(msgCh1, msgCh2)
+	requestMiddleware := middleware.NewMiddlewareManager(
+		msgCh1,
+		msgCh2,
+		helpers.GetEnv("REQUEST_MIDDLEWARES_SOCKET", ""),
+	)
+
+	responseMiddleware := middleware.NewMiddlewareManager(
+		ackCh2,
+		ackCh1,
+		helpers.GetEnv("RESPONSE_MIDDLEWARES_SOCKET", ""),
+	)
 
 	go reader.Start()
-	go middleware.Start()
+	go requestMiddleware.Start()
+	go responseMiddleware.Start()
```

Comment:
Shouldn't the response be initiated after the writer?

Reply:
I don't believe is needed, since the response middleware will wait for messages the writer have to send, then do whatever it has to do and finally send the data to the reader.